components/turret.py

```diff
@@ -147,6 +149,12 @@ def on_enable(self) -> None:
                 self._enable_index_interrupts()
 
     def execute(self) -> None:
+        if self.disable_when_done and self._motor_is_finished():
+            self.disabled = True
+            self.disable_when_done = False
+            self.motor.stopMotor()
+            self.parked = True
+            return
```

Comment:
Why not let the below if statement stop the motor?
```suggestion
            self.parked = True
```